test/controllers/user.js

```diff
@@ -181,110 +207,152 @@ describe('User API calls', function () {
+    });
   });
 
-  it('it should create a Clear token to restore password', function (done) {
-    this.timeout(5000); //to await the email server process
-    request(server)
-      .post('/user/forgot')
-      .set('Accept', 'application/json')
-      .expect('Content-Type', /json/)
-      .send(helper.userData)
-      .expect(200)
-      .end(function (err, res) {
-        if (err) done(err);
-        const userAndToken = res.body;
-        userAndToken.should.be.a('object').with.any.keys('userid', 'token');
-        helper.userForgotPassword.userid = res.body.userid;
-        helper.userForgotPassword.token = res.body.token;
-        done();
+  describe('POST /user/forgot', function () {
+    afterEach(async function () {
+      await helper.deleteMochaUser();
+    });
+
+    it('it should NOT create a Clear token to restore password for a wrong email', async function () {
+      let wrongUserEmail = 'wrong_email@wrong.com';
+      const res = await request(server)
+        .post('/user/forgot')
+        .send(wrongUserEmail)
+        .expect('Content-Type', /json/)
+        .expect(404);
+
+      res.body.message.should.be.a('string');
+    });
+
+    it('it should create a Clear token to restore password', async function () {
+      const user = await helper.prepareUser(server, {
+        role: 'user',
+        email: helper.userData.email,
       });
+      const res = await request(server)
+        .post('/user/forgot')
+        .set('Accept', 'application/json')
+        .expect('Content-Type', /json/)
+        .send({ email: helper.userData.email })
+        .expect(200);
+
+      const userAndUrl = res.body;
+      userAndUrl.should.be
+        .a('object')
+        .with.all.keys('success', 'userid', 'url', 'message');
+    });
   });
+  describe('POST /user/store-password', function () {
+    afterEach(async function () {
+      await helper.deleteMochaUser();
+    });
 
-  it('it should NOT allows to store a new password without a verification url.', function (done) {
-    let wrongVerificationUrl = { ...helper.userForgotPassword };
-    wrongVerificationUrl.token = '';
-    request(server)
-      .post('/user/store-password')
-      .set('Accept', 'application/json')
-      .expect('Content-Type', /json/)
-      .send(wrongVerificationUrl)
-      .expect(200)
-      .end(function (err, res) {
-        if (err) done(err);
-        res.body.message.should.be.a('string');
-        done();
+    it('it should NOT allows to store a new password without a verification url.', async function () {
+      const user = await helper.prepareUser(server, {
+        role: 'user',
+        email: helper.userData.email,
       });
-  });
+      const userStorePassword = { ...user, password: 'newPassword' };
+
+      const res = await request(server)
+        .post('/user/store-password')
+        .set('Accept', 'application/json')
+        .expect('Content-Type', /json/)
+        .send(userStorePassword)
+        .expect(500);
```

Comment:
The previous version checked for 200. Why was it passing?

Reply:
good eye! it's a bug. you can change the password of everybody with only knowing the email.
Not only that, if not, also if this bug is resolved the tokenurl is redeable in the response.
![image](https://user-images.githubusercontent.com/21298844/114641605-cf392680-9ca8-11eb-8352-1adc549c3402.png)
![image](https://user-images.githubusercontent.com/21298844/114641681-f3950300-9ca8-11eb-94d4-1b526896b28b.png)